File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.php
class Solution {

    /**
     * @param String $colors
     * @param Integer[] $neededTime
     * @return Integer
     */
    function minCost($colors, $neededTime) {
        $l = strlen($colors);
        $lc = $colors[0];
        $lt = $neededTime[0];
        $res = 0;
        for($i = 1;$i < $l;$i++) {
            if($lc == $colors[$i]) {
                if($lt <= $neededTime[$i]) {
                    $res += $lt;
                }
                else {
                    $res += $neededTime[$i];
                    continue;
                }
            }
            $lc = $colors[$i];
            $lt = $neededTime[$i];
        }
        return $res;
    }
}

File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.py
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        res = 0
        for c,t in zip(colors,neededTime):
            try:
                if c == lc:
                    if t < lt:
                        res += t
                        continue
                    else:
                        res += lt
                lc,lt = c,t
            except NameError:
                lc,lt = c,t
        return res